src/lib/version/px_update_git_header.py

```diff
@@ -25,48 +35,64 @@
     git_branch_name = ''
 git_version_short = git_version[0:16]
 
-if (os.path.exists('platforms/nuttx/NuttX/nuttx')):
-	nuttx_git_tag = subprocess.check_output('git describe --always --tags --match nuttx-*  --dirty'.split(),
-		                          cwd='platforms/nuttx/NuttX/nuttx', stderr=subprocess.STDOUT).decode('utf-8').strip().replace("nuttx-","v")
-	nuttx_git_tag = re.sub('-.*','.0',nuttx_git_tag)
-	nuttx_git_version = subprocess.check_output('git rev-parse --verify HEAD'.split(),
-		                              cwd='platforms/nuttx/NuttX/nuttx', stderr=subprocess.STDOUT).decode('utf-8').strip()
-	nuttx_git_version_short = nuttx_git_version[0:16]
-else:
-	nuttx_git_version = "0000000"
-	nuttx_git_version_short = "0000000"
-	nuttx_git_tag = "v0.0.0"
-
-mavlink_git_version = subprocess.check_output('git rev-parse --verify HEAD'.split(),
-                                      cwd='mavlink/include/mavlink/v2.0', stderr=subprocess.STDOUT).decode('utf-8').strip()
-mavlink_git_version_short = mavlink_git_version[0:16]
-
-# Generate the header file content
-header = """
-/* Auto Magically Generated file */
-/* Do not edit! */
-#pragma once
+header += """
 #define PX4_GIT_VERSION_STR  "{git_version}"
 #define PX4_GIT_VERSION_BINARY 0x{git_version_short}
 #define PX4_GIT_TAG_STR  "{git_tag}"
 #define PX4_GIT_BRANCH_NAME  "{git_branch_name}"
+""".format(git_tag=git_tag,
+           git_version=git_version,
+           git_version_short=git_version_short,
+           git_branch_name=git_branch_name)
+
+
+# ECL
+if (os.path.exists('src/lib/ecl/.git')):
```

Comment:
What if it doesn't exist?

Reply:
Then it's not defined in the header and the function returns NULL. Eventually I'd like to have these things as proper cmake libraries that export versioning information themselves if found.